pkg/iosrc/uri.go

```diff
@@ -53,9 +56,12 @@ func (p URI) AppendPath(elem ...string) URI {
 	return p
 }
 
-func (p URI) String() string {
-	u := url.URL(p)
-	return (&u).String()
+func (u URI) String() string {
+	if u.IsZero() {
+		return ""
+	}
+	url := url.URL(u)
+	return (&url).String()
```

Comment:
The if statement isn't necessary. (Tests pass with the original implementation.)